packages/client/src/components/RouteLink.ts

```diff
@@ -23,72 +23,75 @@ const guardEvent = (event: MouseEvent): boolean | void => {
   return true
 }
 
-export interface RouteLinkProps extends HTMLAttributes {
-  /**
-   * Whether the link is active to have an active class
-   *
-   * Notice that the active status is not automatically determined according to the current route.
-   *
-   * @default false
-   */
-  active?: boolean
-
-  /**
-   * The class to add when the link is active
-   *
-   * @default 'route-link-active'
-   */
-  activeClass?: string
-
-  /**
-   * The route path to link to
-   */
-  to: string
-}
-
 /**
  * Component to render a link to another route.
  *
  * It's similar to `RouterLink` in `vue-router`, but more lightweight.
  *
  * It's recommended to use `RouteLink` in VuePress.
  */
-export const RouteLink: FunctionalComponent<
-  RouteLinkProps,
-  Record<never, never>,
-  {
-    default: () => string | VNode | (string | VNode)[]
-  }
-> = (
-  { active = false, activeClass = 'route-link-active', to, ...attrs },
-  { slots },
-) => {
-  const router = useRouter()
-  const resolvedPath = resolveRoutePath(to)
+export const RouteLink = defineComponent({
+  name: 'RouteLink',
+
+  props: {
+    /**
+     * The route path to link to
+     */
+    to: {
+      type: String,
+      required: true,
+    },
 
-  const path =
-    // only anchor or query
-    resolvedPath.startsWith('#') || resolvedPath.startsWith('?')
-      ? resolvedPath
-      : withBase(resolvedPath)
+    /**
+     * Whether the link is active to have an active class
+     *
+     * Notice that the active status is not automatically determined according to the current route.
+     */
+    active: Boolean,
 
-  return h(
-    'a',
-    {
-      ...attrs,
-      class: ['route-link', { [activeClass]: active }],
-      href: path,
-      onClick: (event: MouseEvent = {} as MouseEvent) => {
-        guardEvent(event) ? router.push(to).catch() : Promise.resolve()
-      },
+    /**
+     * The class to add when the link is active
+     */
+    activeClass: {
+      type: String,
+      default: 'route-link-active',
     },
-    slots.default?.(),
-  )
-}
+  },
 
-RouteLink.displayName = 'RouteLink'
-RouteLink.props = {
-  active: Boolean,
-  activeClass: String,
-  to: String,
-}
+  slots: Object as SlotsType<{
+    default: () => string | VNode | (string | VNode)[]
+  }>,
+
+  setup(props, { slots }) {
+    const router = useRouter()
+    const route = useRoute()
+
+    const path = computed(() => {
+      if (
+        // as is
+        ['#', '?', 'http://', 'https://', '//'].some((prefix) =>
```

Comment:
Maybe extract to a util function? Or at least extract the array to a constant.

Also need test cases for links with protocols